src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -220,6 +220,16 @@
              StopOnFirstFailure="true" />
   </Target>
 
+  <!-- Update build command with repo-conditioned arguments. -->
+  <Target Name="UpdateBuildCommandWithRepoConditionedArguments"
+          BeforeTargets="Build">
+    <PropertyGroup Condition="'$(UseInnerClone)' == 'true'">
+      <BuildCommand>$(BuildCommand) /p:UseInnerClone=true</BuildCommand>
+      <BuildCommand>$(BuildCommand) /p:CopySrcInsteadOfClone=true</BuildCommand>
+      <BuildCommand>$(BuildCommand) /p:CopyWipIntoInnerSourceBuildRepo=true</BuildCommand>
+    </PropertyGroup>
+  </Target>
```

Comment:
This shouldn't be a target and you shouldn't need to mutate `BuildCommand`. Just append to `BuildArgs` before `BuildArgs` is written into `BuildCommand`.

Reply:
I can modify this to append to `BuildArgs`. Are you suggesting that I add these property assignments right before this line? https://github.com/dotnet/installer/blob/9b36bbc4ab15dd844a358551278f6b8467c64641/src/SourceBuild/content/repo-projects/Directory.Build.targets#L6

They can't be in `Directory.Build.props` as all of those are evaluated before any property in project file (i.e. the new `UseInnerClone`)

Reply:
Yes exactly before that line.

Reply:
Fixed with https://github.com/dotnet/installer/pull/18307/commits/41fd7e996f4a6555a8d9a544101ecb87e1092138